addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,307 @@
+/**
+ * @author RT_Borg
+ *
+ * Uses topic id as a proxy for age, which doesn't require loading other pages
+ * to read actual dates.
+ */
+
+export default async function ({ addon, global, console, msg }) {
+  // Default is a little over a month's worth of topics. Reasonably stable through 2022.
+  // There's a lot of tolerance. Half or twice as many topics filter pretty much the same.
+  const TOPICS_PER_MONTH = 7500;
+  const defaultForums = [
+    "New Scratchers",
+    "Help with Scripts",
+    "Questions about Scratch",
+    "Bugs and Glitches",
+    "Project Ideas",
+  ];
+  // Forums beyond these seem very poor options,
+  // as many posts live long or are appropriately revived
+  const customForumSettingToForum = {
+    applyToNewScratchers: "New Scratchers",
+    applyToHelpWithScripts: "Help with Scripts",
+    applyToQuestionsAboutScratch: "Questions about Scratch",
+    applyToBugsAndGlitches: "Bugs and Glitches",
+    applyToProjectIdeas: "Project Ideas",
+    applyToRequests: "Requests",
+    applyToSuggestions: "Suggestions",
+    applyToOpenSourceProjects: "Open Source Projects",
+  };
+  const searchResultsPageName = "Search Results"; // recent posts and unanswered posts
+  // mobile urls look like: https://scratch.mit.edu/discuss/m/6/*
+  const isMobileSite = /^\/discuss\/m\//.test(location.pathname);
+  // tableBodyNode will be null on mobile and non-table search page:
+  // https://scratch.mit.edu/discuss/search/?action=show_user&show_as=posts
+  const tableBodyNode = document.querySelector("tbody"); // null on mobile
+
+  if (!tableBodyNode && !isMobileSite) {
+    // Standard site page without a table. Don't run on this page.
+    return;
+  }
```

Comment:
Not important, but conditions like this can be condensed into a single line.
```suggestion
  // Standard site page without a table. Don't run on this page.
  if (!tableBodyNode && !isMobileSite) return;
```